website/src/preferences-context.tsx

```diff
@@ -15,51 +15,89 @@ export const usePreferences = () => useContext(PreferencesContext)
 
 export const PreferencesProvider = (props: any) => {
   // Some preferences hooks setup
-  const [viewMode, setViewMode] = useState(savedViewMode())
+  const [levelOfDetail, setLevelOfDetail] = useState(savedLevelOfDetail())
   const [cherokeeRepresentation, setCherokeeRepresentation] = useState(
     savedCherokeeRepresentation()
   )
+
+  useEffect(() => {
+    save(PreferenceKey.LevelOfDetail, levelOfDetail)
+    save(PreferenceKey.CherokeeRepresentation, cherokeeRepresentation)
+    // Remove all deprecated cookies, keep this code for at least several months
+    // to let users clear their site cookies.
+    Cookies.remove(DeprecatedCookie.LevelOfDetail)
+    Cookies.remove(DeprecatedCookie.Romanization)
+  }, [levelOfDetail, cherokeeRepresentation])
+
   return (
     <PreferencesContext.Provider
       value={{
-        viewMode,
-        setViewMode: persistLocally(PreferenceKey.ViewMode, setViewMode),
+        levelOfDetail,
+        setLevelOfDetail,
         cherokeeRepresentation,
-        setCherokeeRepresentation: persistLocally(
-          PreferenceKey.CherokeeRepresentation,
-          setCherokeeRepresentation
-        ),
+        setCherokeeRepresentation,
       }}
     >
       {props.children}
     </PreferencesContext.Provider>
   )
 }
 
-function persistLocally<T extends { toString: () => string }>(
-  key: PreferenceKey,
-  setValue: (value: T) => void
-) {
-  return function (value: T) {
-    Cookies.set(key, value.toString(), {
-      sameSite: "strict",
-      secure: true,
-    })
-    return setValue(value)
-  }
+function save(key: PreferenceKey, value: any) {
+  LocalStorage.setItem(key, value.toString())
 }
 
 // Avoid changing these keys at all costs, because that will essentially reset
 // saved user preferences.
 enum PreferenceKey {
-  ViewMode = "experienceLevel",
-  CherokeeRepresentation = "cherokeeSystem",
+  LevelOfDetail = "level-of-detail",
+  CherokeeRepresentation = "cherokee-system",
 }
 
-const savedViewMode = () =>
-  Number.parseInt(Cookies.get(PreferenceKey.ViewMode) ?? "1") as ViewMode
+enum DeprecatedCookie {
+  // TODO Remove this after several months, giving users time to have their
+  // settings migrated.
+  LevelOfDetail = "experienceLevel",
+  Romanization = "phonetics",
+}
+
+const LocalStorage =
+  typeof window !== "undefined"
+    ? window.localStorage
+    : { getItem: () => null, setItem: () => {} }
+
+const savedLevelOfDetail = (): LevelOfDetail => {
+  const newCookie = LocalStorage.getItem(PreferenceKey.LevelOfDetail)
```

Comment:
It might be confusing referring to this as a cookie, since a big part of this change is switching over to local storage from cookies. Maybe referring to them as `newData` and `oldData`, or even just `stored` and `cookie`.

Reply:
Oh yeah, definitely. I wrote the code before using local storage for the new value. Will change.